src/components/tooltip/tooltip.test.js

```diff
@@ -0,0 +1,175 @@
+        <Tooltip {...props} message="I am a tooltip!" isVisible>
+          <Component>target</Component>
+        </Tooltip>
+      </div>
+    </>
+  );
+};
+
+context("Tests for Tooltip component", () => {
+  describe("should check Tooltip component properties", () => {
+    it.each(testData)(
+      "should check %s as message for Tooltip component",
+      (message) => {
+        CypressMountWithProviders(<Tooltip isVisible message={message} />);
+        tooltipPreview().should("have.text", message);
+      }
+    );
+
+    it.each(testData)("should check %s as Id for Tooltip component", (id) => {
+      CypressMountWithProviders(<TooltipComponent id={id} />);
+      tooltipPreview().should("have.id", id);
+    });
+
+    it.each([
+      [true, "be.visible"],
+      [false, "not.exist"],
+    ])(
+      "should check when tooltip visibility is %s for Tooltip component",
+      (bool, state) => {
+        CypressMountWithProviders(<Tooltip isVisible={bool} />);
+        tooltipPreview().should(state);
+      }
+    );
+
+    it.each(["bottom", "left", "right", "top"])(
+      "should check %s position of tooltip for Tooltip component",
+      (position) => {
+        CypressMountWithProviders(
+          <TooltipComponent position={position} isVisible>
+            {`This tooltip is positioned ${position}`}
+          </TooltipComponent>
+        );
+        tooltipPreview().should("be.visible").and("have.css", position);
+      }
+    );
+
+    it.each(["undefined", "error"])(
+      "should check %s type for Tooltip component",
+      (type) => {
+        CypressMountWithProviders(<TooltipComponent type={type} />);
+        tooltipPreview().should("have.attr", "type", type);
+      }
+    );
+
+    it.each([
+      ["orange", "rgb(255, 156, 75)"],
+      ["red", "rgb(205, 56, 75)"],
+      ["black", "rgb(0, 0, 0)"],
+      ["brown", "rgb(105, 61, 57)"],
+    ])(
+      "should check tooltip background-color as %s for Tooltip component",
+      (name, color) => {
+        CypressMountWithProviders(<TooltipComponent bgColor={color} />);
+        tooltipPreview().should("have.css", "background-color", color);
+      }
+    );
+
+    it.each([
+      ["orange", "rgb(255, 156, 75)"],
+      ["red", "rgb(205, 56, 75)"],
+      ["black", "rgb(0, 0, 0)"],
+      ["brown", "rgb(105, 61, 57)"],
+    ])(
+      "should check tooltip font color as %s for Tooltip component",
+      (name, color) => {
+        CypressMountWithProviders(<TooltipComponent fontColor={color} />);
+        tooltipPreview().should("have.css", "color", color);
+      }
+    );
+
+    it.each(["top", "bottom", "right", "left"])(
+      "should check flip position to the %s for Tooltip component",
+      (position) => {
+        CypressMountWithProviders(
+          <TooltipComponent flipOverrides={[position]}>
+            {`This tooltip is positioned ${position}`}
+          </TooltipComponent>
+        );
+        cy.viewport(700, 120);
+        cy.scrollTo(50, 50);
```

Comment:
I think that this test is written wrong, we need to have different way to scroll, for ex. if we are checking that the component will be shown to the left we limit the space to right and so on

Reply:
This same behaviour I have tested for Help component using this same logic.